src/Bellatrix.Web/services/DevToolsService.cs

```diff
@@ -28,6 +30,99 @@ public DevToolsService(IWebDriver wrappedDriver)
+        networkInterceptor.NetworkResponseReceived += (o, s) =>
+        {
+            ResponsesHistory.Add(s);
+        };
+        networkInterceptor.NetworkRequestSent += (o, s) =>
+        {
+            RequestsHistory.Add(s);
+        };
+
+        networkInterceptor.StartMonitoring();
+    }
+
+    public void ClearRequestsResponsesHistory()
+    {
+        RequestsHistory.Clear();
+        ResponsesHistory.Clear();
+    }
+
+    public List<string> GetSpecificRequestUrls(string requestName)
+    {
+        HashSet<NetworkRequestSentEventArgs> requests = new HashSet<NetworkRequestSentEventArgs>(RequestsHistory);
+        return requests.ToList()
+            .FindAll(r => r.RequestUrl.ToString().Contains(requestName))
+            .Select(fr => fr.RequestUrl).ToList();
+    }
+
+    public void OverrideScreenResolution(long screenHeight, long screenWidth, bool mobile, double deviceScaleFactor)
+    {
+        var settings = new SetDeviceMetricsOverrideCommandSettings();
+        settings.ScreenHeight = screenHeight;
+        settings.ScreenWidth = screenWidth;
+        settings.Mobile = mobile;
+        settings.DeviceScaleFactor = deviceScaleFactor;
+
+        DevToolsSession.SendCommand(settings);
+    }
+
+    public long GetResponseContentLengthByPartialUrl(string partialUrl)
+    {
+        var exactResponse = ResponsesHistory.ToList().Find(r => r.ResponseUrl.Contains(partialUrl)).ResponseHeaders["content-length"].ToString();
+
+        return exactResponse.ToLong();
+    }
+
+    public string GetResponseContentTypeByPartialUrl(string partialUrl)
+    {
+        var exactResponse = ResponsesHistory.ToList().Find(r => r.ResponseUrl.Contains(partialUrl)).ResponseHeaders["content-type"].ToString();
+
+        return exactResponse;
+    }
+
+    public void AssertResponse404ErrorCodeRecievedByPartialUrl(string partialUrl)
+    {
+        var response = ResponsesHistory.ToList().Find(r => r.ResponseUrl.Contains(partialUrl)).ResponseStatusCode;
+
+        Assert.AreEqual(response, 404, "404 Error code not detected on the page.");
+    }
+
+    public void AssertNoErrorCodes()
+    {
+        bool areThereErrorCodes = ResponsesHistory.Any(r => r.ResponseStatusCode > 400 && r.ResponseStatusCode < 599);
+        Assert.IsFalse(areThereErrorCodes, "Error codes detected on the page.");
+    }
+
+    public void AssertRequestMade(string url)
+    {
+        bool areRequestsMade = ResponsesHistory.Any(r => r.ResponseUrl.Contains(url));
+        Assert.IsTrue(areRequestsMade, $"Request {url} was not made.");
+    }
+
+    public void AssertRequestNotMade(string url)
+    {
+        bool areRequestsMade = ResponsesHistory.Any(r => r.ResponseUrl.Contains(url));
+        Assert.IsFalse(areRequestsMade, $"Request {url} was made.");
+    }
+
+    public int CountRequestsMadeByFileFormat(string fileFormat)
+    {
+        var exactResponse = ResponsesHistory.ToList().FindAll(r => r.ResponseUrl.Contains(fileFormat)).ToList();
```

Comment:
I guess you should use EndsWith instead of Contains?

Reply:
Fixed